lib/diagnostic.js

```diff
@@ -32,14 +40,26 @@ const any = function any(array, callback) {
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 const select = function select(array, callback) {
-
+  let newArray = [];
+  for (let i = 0; i < array.length; i++) {
+    if (callback(array[i])) {
+      newArray.push(callback(array[i]));
+      return newArray;
```

Comment:
Whoops... we don't want to put the return statement there. If we do, it will return the new array after adding the first element - we will never get back more than the first matching result.